ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/sse/ServerSentEvents.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.plugins.sse
+
+import io.ktor.client.*
+import io.ktor.client.engine.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.util.*
+import io.ktor.util.logging.*
+
+internal val LOGGER = KtorSimpleLogger("io.ktor.client.plugins.sse.ServerSentEvents")
+
+/**
+ * Indicates if a client engine supports Server-sent events.
+ */
+public object ServerSentEventsCapability : HttpClientEngineCapability<Unit> {
+    override fun toString(): String = "ServerSentEventsCapability"
+}
+
+/**
+ * Client Server-sent events plugin.
+ *
+ * @property reconnectionTimeMillis - time for client to wait before attempting to reconnect.
+ */
+public class ServerSentEvents(
+    public val reconnectionTimeMillis: Long,
+) {
+    /**
+     * [ServerSentEvents] configuration.
+     */
+    @KtorDsl
+    public class Config {
+        /**
+         * The reconnection time. If the connection to the server is lost,
+         * the client will wait for the specified time before attempting to reconnect.
+         * Note that this parameter is not supported for some engines.
+         */
+        public var reconnectionTimeMillis: Long = 1000
```

Comment:
may be also add support for kotlin.time.Duration?

Reply:
```kotlin
public var reconnectionTime: Duration
        get() = reconnectionTimeMillis.milliseconds
        set(value) {
            reconnectionTimeMillis = value.inWholeMilliseconds
        }
```

Do you mean to add also something like this?

Reply:
yes, or even leave only type safe way via `Duration`. It depends on what is the overall strategy in ktor regarding `millis` properties. There are a lot of them in different plugins. Somewhere there are extensions for `Duration`, somewhere not.
I believe that 3.0 is upcoming release and `kotlin.time.Duration` is already Stable, may be it's a good idea to check all plugins (server and client) and may be other places, there `millis` properties are used and add `Duration` counterpart.